### PR TITLE
Moving to llvm commit 0128f8016770655fe7a40d3657f00853e6badb93

### PR DESCRIPTION
**Pull request by nbpatel**
Please review these guidelines to help with the review process:
- [x] Have you provided a meaningful PR description?
- [x] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] If this PR is a work in progress, are you filing the PR as a draft?
- [x] Have you organized your commits logically and ensured each can be built by itself?

Creating a PR on behalf of Abdhulla to update the LLVM $SHA.

### CODE REVIEW

**Comment by fschlimb** [comment]
@Hardcode84

**Review by fschlimb — APPROVED** [review]

**Comment by Hardcode84** [comment]
LGTM to me, I'll update main after this is merged

**Pull request merged**
